app/models/service_provider_identity.rb

```diff
@@ -66,9 +71,15 @@ def happened_at
   end
 
   def email_address_for_sharing
```

Comment:
Related to my previous comment, my impression is that we wouldn't change the retrieval of `email_address_for_sharing`, we'd just ensure that the identity's associated `email_address_id` database field isn't populated unless the `verified_attributes` array includes `email` (and not `all_emails`).

We might also want to account for a scenario where a partner initially requests a single email and then later requests all emails. In that case, we might want to explicitly unset the associated `email_address_id` (especially since they won't be able to change it after #11701). Maybe we could use an [Active Record callback](https://guides.rubyonrails.org/active_record_callbacks.html#available-callbacks) to help manage this.